install_config/web_console_customization.adoc

```diff
@@ -172,6 +172,142 @@ assetConfig:
     - /path/to/cli-links.js
 ----
 ====
+
+[[adding-to-or-changing-navigation-menus]]
+=== Adding to or Changing Navigation Menus
+
+==== Top Navigation Dropdown Menus
+
+In the top navigation bar of the console there are two menu dropdowns, the help icon menu
+and the user menu.  Each of these menus can have additional menu items added to them.  They use
+the link:https://github.com/openshift/angular-extension-registry[angular-extension-registry].  A basic
+example of extending a menu is shown below, where 'myExtensionModule' is a placeholder and should be
+unique enough that it does not clash with any angular modules in the future.
+
+====
+----
+angular
+  .module('myExtensionModule', ['openshiftConsole'])
+  .run([
+    'extensionRegistry',
+    function(extensionRegistry) {
+      extensionRegistry
+        .add('nav-help-dropdown', function() {
+          return [
+            {
+              type: 'dom',
+              node: '<li><a href="http://www.example.com/report" target="_blank">Report a Bug</a></li>'
+            }, {
+              type: 'dom',
+              node: '<li class="divider"></li>'  // If you want a horizontal divider to appear in the menu
+            }, {
+              type: 'dom',
+              node: '<li><a href="http://www.example.com/status" target="_blank">System Status</a></li>'
+            }
+          ];
+        });
+    }
+  ]);
+
+hawtioPluginLoader.addModule('myExtensionModule');
+----
+====
+
+The available extension points are:
+
+* nav-help-dropdown - the help icon dropdown menu, visible at desktop screen widths
+* nav-user-dropdown - the user dropdown menu, visible at desktop screen widths
+* nav-dropdown-mobile - the single menu for top navigation items at mobile screen widths
+
+==== Project Left Navigation
+
+When you are navigating within a project, a menu appears on the left with primary and secondary navigation.
+This menu structure is defined as a constant and can be overridden or modified.
+
+[NOTE]
+====
+Significant customizations to the project navigation may affect the user experience and should be done with
+careful consideration. You may need to update this customization in future upgrades if you modify existing
+navigation items.
+====
+
+====
+----
+// Append a new primary nav item.  This is a simple direct navigation item
+// with no secondary menu.
+window.OPENSHIFT_CONSTANTS.PROJECT_NAVIGATION.push({
+  label: "Overview",            // The text label
+  iconClass: "fa fa-dashboard", // The icon you want to appear
+  href: "/overview"             // Where to go when this nav item is clicked
+});
+
+// Splice a primary nav item to a specific spot in the list.  This primary item has
+// a secondary menu.
+window.OPENSHIFT_CONSTANTS.PROJECT_NAVIGATION.splice(2, 0, { // Insert at the third spot
+  label: "Applications",
```

Comment:
Is it confusing for the example to add a menu item that already exists? Looks like you might be modifying the existing one or adding a secondary item to it.